### ISSUE TITLE
Version 1.4.1 breaks pseudo parameters

### ISSUE THREAD

**Issue by jquatier** [horike37/serverless-step-functions#119]
# This is a Bug Report

## Description

The most recent version of this plugin breaks the functionality of using serverless-pseudo-parameters (https://www.npmjs.com/package/serverless-pseudo-parameters)

```
Serverless Error ---------------------------------------

  Invalid variable reference syntax for variable AWS::Region. You can only reference env vars, options, & files. You can check our docs for more info.
```
Note: Rolling back to 1.4.0 deploys successfully.

Configuration:
```
stepFunctions:
  stateMachines:
    myStateMachine:
      definition:
        StartAt: FirstState
        States:
          FirstState:
            Type: Task
            Resource: "arn:aws:lambda:#{AWS::Region}:#{AWS::AccountId}:function:firstState"
            Next: NextState
```

## Additional Data
* ***Serverless Framework Core Version you're using***: 1.23.0
* ***The Plugin Version you're using***: 1.4.1
* ***Operating System***: Mac OSX / Ubuntu

**Comment by horike37** [comment]
@jquatier 
Thank you for the reporting :+1: 
Will take a look it

**Comment by horike37** [comment]
@jquatier
I tried to reproduce it. unfortunately, I couldn't get the error with the version.
Is your project hosted on GitHub or any other public places? If so, I would like to try it again with your project.

**Comment by theburningmonk** [comment]
I have run into a similar issue with this as well, not sure if it's related, but it only happens with the latest version 1.4.1. The error I'm getting is different: `Invalid variable reference syntax for variable AWS::Partition. You can only reference env vars, options, & files. You can check our docs for more info.`

But interestingly, it only happens when I install SLS as a dev dependency and run `deploy` against that. I have the same version of SLS installed on my machine and it works just fine when I run `deploy` with the local machine version of SLS.

**Comment by msl-kabo** [comment]
Same issue here. `serverless` 1.27.2 and `serverless-step-functions` installed as dev dependencies. Downgrading to 1.4.0 works.

**Comment by horike37** [comment]
Thank you for reporting @theburningmonk and @msl-kabo :+1: 
Is it possible to share serverless.yml you are using? I would like to try to reproduce it.

@theburningmonk
How does it work with version 1.4.0? I would like to know whether to be a specific problem of v1.4.1

**Comment by theburningmonk** [comment]
In my case the problems seems to be specific to v1.4.1. Rolling back to v1.4.0 worked for me.

@horike37 do you need a minimal example to reproduce the issue? I can put something together

**Comment by horike37** [comment]
>do you need a minimal example to reproduce the issue?
Yes, please!

**Comment by ColeTMNZ** [comment]
Adding my two cents as I've spent the last few hours chasing this issue. 

I followed the error, and found it is generated by Serverless: https://github.com/serverless/serverless/blob/a2ae70b97381c3595b3888598ba6a36535593854/lib/classes/Variables.js#L489

The Error occurs when serverless tries to find a corresponding variable to satisfy the following: https://github.com/serverless/serverless/blob/a2ae70b97381c3595b3888598ba6a36535593854/lib/plugins/aws/package/lib/mergeIamTemplates.js#L96 - This dosn't make sense as that line should be parsed by cloudformation, not serverless. 

What makes all this more confusing, is rolling serverless-step-functions back to 1.4.0 works again. (Also removing the step function plugin allows it to work as well. 

Hopefully, this is helpful...

**Comment by ballwood** [comment]
@horike37  I had the same issue. Seems ordering of plugins is the problem, if you put `serverless-step-functions` before `serverless-pseudo-parameters` it works.

**Comment by delkopiso** [comment]
>if you put serverless-step-functions before serverless-pseudo-parameters it works.

This didn't work for me

**Comment by ColeTMNZ** [comment]
I have checked my plugin order, and I've always had `serverless-step-functions` at the top - and was experiencing the issue. So - I concur that order (specifically for this issue) is not a fix.

**Comment by ballwood** [comment]
@ColeTMNZ @delkopiso my bad, just checked and yarn.lock was resolving to an earlier version of step-functions even though I'd performed the 1.4.1 upgrade. Dropping down to 1.4.0 as previously stated fixes it.

**Comment by ColeTMNZ** [comment]
At least we don't have inconsistent results :-D @ballwood

**Comment by rmax** [comment]
I'm having this issue as well. Fixed by going back to `1.4.0`.

**Comment by horike37** [comment]
This problem has been fixed on v1.5.0. Let me know if you still face the same 👍 
Thank all people here for helping it out!
